engine/render/src/render/render.cpp

```diff
@@ -435,19 +454,33 @@ namespace dmRender
             }
 
             // Write z values...
+            int num_visibility_skipped = 0;
             for (uint32_t i = range.m_Start; i < range.m_Start+range.m_Count; ++i)
             {
                 uint32_t idx = context->m_RenderListSortIndices[i];
                 RenderListEntry* entry = &entries[idx];
+                if (entry->m_Visibility == dmRender::VISIBILITY_NONE)
+                {
+                    num_visibility_skipped++;
+                    continue;
+                }
+
                 if (entry->m_MajorOrder != RENDER_ORDER_WORLD)
-                    continue; // Could perhaps break here, if we also sorted on the major order (cost more when I tested it /MAWE)
+                {
+                     continue; // Could perhaps break here, if we also sorted on the major order (cost more when I tested it /MAWE)
+                }
 
                 const Vector4 res = transform * entry->m_WorldPosition;
                 const float zw = res.getZ() / res.getW();
                 sort_values[idx].m_ZW = zw;
                 if (zw < minZW) minZW = zw;
                 if (zw > maxZW) maxZW = zw;
```

Comment:
I guess there's also an option to use this Z and compare with the frustum near/far